### PR TITLE
DOC: add reference to poisson rng

### PR DESCRIPTION
**Pull request by juliantaylor**
[ci skip]

as googling numeric constants should not be the way to find the description of the code.

### CODE REVIEW

File: numpy/random/mtrand/distributions.c [hunk 1]
```diff
@@ -500,6 +500,10 @@ long rk_poisson_mult(rk_state *state, double lam)
     }
 }
 
+/*
```
Comment (charris) [inline numpy/random/mtrand/distributions.c]: Google shows a more complete citation.

W. Hörmann and G. Derflinger
The transformed rejection method for generating random variables, an alternative to the ratio of uniforms method
Communications in Statistics: Simulation and Computation 23(3), 847-860 (1994)

Comment (juliantaylor) [inline numpy/random/mtrand/distributions.c]: thats another paper by the same author, likely generalized
the exact algorithm was first published in W. Hörmann 
The transformed rejection method for generating Poisson random variables Insurance: Mathematics and Economics 12, 39-45 (1993)
I kept it short as you can find the thing by googling the numbers involved so I though the title ie enough ;)

Comment (juliantaylor) [inline numpy/random/mtrand/distributions.c]: updated with a more complete citation

**Comment by charris** [comment]
Thanks Julian.

**Pull request merged**